### PR TITLE
[fix](bthread) fix bthread hang

### PR DESCRIPTION
**Pull request by Tanya-W**
# Proposed changes
Issue Number: close `problem two` of https://github.com/apache/doris/issues/16606

## Problem summary
- define member variable `_cancelled` in FragmentExecState as atomic

- use `bthread_usleep` instead of `std::this_thread::sleep_for` in function `VNodeChannel::close_wait`

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
3. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
5. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/vec/sink/vtablet_sink.cpp [hunk 1]
```diff
@@ -689,7 +689,8 @@ Status VNodeChannel::close_wait(RuntimeState* state) {
 
     // waiting for finished, it may take a long time, so we couldn't set a timeout
     while (!_add_batches_finished && !_cancelled) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(1));
```
Comment (airborne12) [inline be/src/vec/sink/vtablet_sink.cpp:692]: under bthread/pthread context needed here?

**Review by morningman — DISMISSED** [review]
LGTM

**Review by dataroaring — DISMISSED** [review]
LGTM

File: be/src/vec/sink/vtablet_sink.cpp [hunk 2]
```diff
@@ -688,8 +688,9 @@ Status VNodeChannel::close_wait(RuntimeState* state) {
     }
 
     // waiting for finished, it may take a long time, so we couldn't set a timeout
-    while (!_add_batches_finished && !_cancelled) {
-        std::this_thread::sleep_for(std::chrono::milliseconds(1));
+    while (!_add_batches_finished && !_cancelled && !state->is_cancelled()) {
+        // std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        bthread_usleep(1000);
```
Comment (yiguolei) [inline be/src/vec/sink/vtablet_sink.cpp:693]: Please not use bhtread here, @chenlinzhong will add pthread pool after internal_service.cpp for every brpc request

**Comment by yiguolei** [comment]
Please not merge this PR to master branch. Could merge it to 1.1.6 and 1.2.

File: be/src/vec/sink/vtablet_sink.cpp (hunk 2 above)
Comment (dataroaring) [reply be/src/vec/sink/vtablet_sink.cpp:693]: Here, there is another bug, if a fragment is not open, then close_wait does not handle cancell. bthread_sleep works like sleep in pthread. so I merged it.

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**